x/staking/side_handler_test.go

```diff
@@ -0,0 +1,1460 @@
+package staking_test
+
+import (
+	"math/big"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/maticnetwork/heimdall/x/staking/test_helper"
+
+	checkPointSim "github.com/maticnetwork/heimdall/x/checkpoint/simulation"
+
+	"github.com/maticnetwork/bor/common"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	ethTypes "github.com/maticnetwork/bor/core/types"
+	"github.com/maticnetwork/heimdall/app"
+	"github.com/maticnetwork/heimdall/contracts/stakinginfo"
+	"github.com/maticnetwork/heimdall/helper"
+	"github.com/maticnetwork/heimdall/helper/mocks"
+	hmTypes "github.com/maticnetwork/heimdall/types"
+	hmCommonTypes "github.com/maticnetwork/heimdall/types/common"
+	"github.com/maticnetwork/heimdall/x/staking"
+	stakingSim "github.com/maticnetwork/heimdall/x/staking/simulation"
+	"github.com/maticnetwork/heimdall/x/staking/types"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"github.com/tendermint/tendermint/crypto/secp256k1"
+	tmprototypes "github.com/tendermint/tendermint/proto/tendermint/types"
+)
+
+type SideHandlerTestSuite struct {
+	suite.Suite
+
+	app            *app.HeimdallApp
+	ctx            sdk.Context
+	sideHandler    hmTypes.SideTxHandler
+	postHandler    hmTypes.PostTxHandler
+	contractCaller mocks.IContractCaller
+	r              *rand.Rand
+}
+
+func (suite *SideHandlerTestSuite) SetupTest() {
+	suite.app, suite.ctx, _ = test_helper.CreateTestApp(false)
+
+	suite.contractCaller = mocks.IContractCaller{}
+	suite.sideHandler = staking.NewSideTxHandler(suite.app.StakingKeeper, &suite.contractCaller)
+	suite.postHandler = staking.NewPostTxHandler(suite.app.StakingKeeper, &suite.contractCaller)
+
+	// random generator
+	s1 := rand.NewSource(time.Now().UnixNano())
+	suite.r = rand.New(s1)
+}
+
+func TestSideHandlerTestSuite(t *testing.T) {
+	suite.Run(t, new(SideHandlerTestSuite))
+}
+
+// Test Cases
+
+func (suite *SideHandlerTestSuite) TestSideHandler() {
+	t, ctx := suite.T(), suite.ctx
+
+	// side handler
+	result := suite.sideHandler(ctx, nil)
+	require.Equal(t, result.Code, uint32(6))
```

Comment:
can we use a constant?

Reply:
yes we can

Reply:
Let's use:
`require.Equal(t, sdkerrors.ErrUnknownRequest.ABCICode(), result.Code)
`